src/test/java/uk/gov/justice/digital/oasys/util/KeyGeneration.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.justice.digital.oasys.util;
+
+import com.nimbusds.jose.JOSEException;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSObject;
+import com.nimbusds.jose.JWSSigner;
+import com.nimbusds.jose.Payload;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.jwk.RSAKey;
+import com.nimbusds.jose.jwk.gen.RSAKeyGenerator;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.text.ParseException;
+
+/**
+ * Tool used to generate key pairs and signed tokens for use in the integration tests
+ */
+public class KeyGeneration {
```

Comment:
For the future - I think we should probably move this utility out to a separate library.